src/main/java/za/co/neslotech/spring/trainspringbootinsight/service/BudgetService.java

```diff
@@ -0,0 +1,56 @@
+package za.co.neslotech.spring.trainspringbootinsight.service;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.stereotype.Service;
+import za.co.neslotech.spring.trainspringbootinsight.model.Budget;
+import za.co.neslotech.spring.trainspringbootinsight.repository.IBudgetRepository;
+
+import java.util.List;
+
+@Service
+public class BudgetService implements CrudService<Budget, Long> {
```

Comment:
One small improvement you can potentially make in the future is if you explicitly define the ID type to a Long, there would no need to provide the type to your abstract class. It will also ensure that all of your models follow the same standard.

Another improvement to make is to create an Abstract Entity class that all your models extend to further ensure this principal.